apps/dashboard/src/@3rdweb-sdk/react/hooks/useBalanceForAddress.tsx

```diff
@@ -1,13 +1,23 @@
+import { thirdwebClient } from "@/constants/client";
 import { useQuery } from "@tanstack/react-query";
-import { useSDK } from "@thirdweb-dev/react";
+import { defineChain } from "thirdweb";
+import { getWalletBalance } from "thirdweb/wallets";
 import invariant from "tiny-invariant";
 import { walletKeys } from "../cache-keys";
 
-export function useBalanceForAddress(address: string | undefined) {
-  const sdk = useSDK();
+export function useBalanceForAddress(
+  address: string | undefined,
+  chainId: number | undefined,
+) {
   invariant(address, "Address is not provided");
+  invariant(chainId, "chainId is undefined");
   return useQuery(walletKeys.balances(address), async () => {
-    invariant(sdk, "SDK is not initialized");
-    return await sdk.getBalance(address);
+    const chain = defineChain(chainId);
+    const data = await getWalletBalance({
+      chain,
+      client: thirdwebClient,
+      address,
+    });
+    return data;
   });
```

Comment:
can u update this to the more modern `useQuery` notation while you're at it:

```ts
useQuery({
  queryKey: ...,
  queryFn: async () => {...}
})
```

see: https://tanstack.com/query/v4/docs/framework/react/guides/queries